src/common/options.rs

```diff
@@ -112,13 +109,43 @@ impl Endpointing {
     #[allow(missing_docs)]
     pub fn as_str(&self) -> String {
         match self {
-            Endpointing::True(_value) => "true".to_string(),
-            Endpointing::False(_value) => "false".to_string(),
+            Endpointing::Enabled(value) => {
+                if *value {
+                    "true".to_string()
+                } else {
+                    "false".to_string()
+                }
+            },
             Endpointing::CustomValue(value) => format!("{}", value),
         }
     }
 }
 
+#[derive(Debug, PartialEq, Clone)]
+struct Extra {
+    key: String,
+    value: String,
+}
+
+impl Extra {
+    pub fn new(key: &str, value: &str) -> Self {
+        Self {
+            key: key.to_string(),
+            value: value.to_string(),
+        }
+    }
+}
+
+impl Serialize for Extra {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let output = format!("{}:{}", self.key, self.value);
+        serializer.serialize_str(&output)
+    }
+}
+
```

Comment:
```suggestion
impl Serialize for Extra {
    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
    where
        S: serde::Serializer,
    {
        let output = Vec::new();
        for (key, value) in self.iter() {
           output.push(format!("extra={}:{}", key, value));
         }
        serializer.serialize_str(output.join("&");
    }
}
```

When converting to a HashMap, will become something like this.